src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -590,19 +590,16 @@ internal int BytesToWrite
         internal SerialStream(string portName, int baudRate, Parity parity, int dataBits, StopBits stopBits, int readTimeout, int writeTimeout, Handshake handshake,
             bool dtrEnable, bool rtsEnable, bool discardNull, byte parityReplace)
         {
-            int flags = NativeMethods.FILE_FLAG_OVERLAPPED;
-
-            if ((portName == null) || !portName.StartsWith("COM", StringComparison.OrdinalIgnoreCase))
+            if ((portName == null) ||
+                !portName.StartsWith("COM", StringComparison.OrdinalIgnoreCase) || 
+                !uint.TryParse(portName.Substring(3), out uint portNumber))
+            {
                 throw new ArgumentException(SR.Arg_InvalidSerialPort, nameof(portName));
+            }
 
             // Error checking done in SerialPort.
 
-            SafeFileHandle tempHandle = Interop.Kernel32.CreateFileDefaultSecurity(
-                @"\\?\" + portName,
-                Interop.Kernel32.GenericOperations.GENERIC_READ | Interop.Kernel32.GenericOperations.GENERIC_WRITE,
-                0,              // comm devices must be opened w/exclusive-access
-                FileMode.Open,  // comm devices must use OPEN_EXISTING
-                flags);
+            SafeFileHandle tempHandle = OpenPort(portNumber);
```

Comment:
It is a good idea, that said, it is unlikely to be a problem since the frequency of opening a port should be small. Let's keep as it is and wait for #22403 so then we keep the (minor) benefit of having consistent exceptions for the same issue.